tap_postgres/db.py

```diff
@@ -59,6 +59,16 @@ def open_connection(conn_config, logical_replication=False):
 
     return conn
 
+def prepare_columns_for_select_sql(c, md_map):
+    column_name = ' "{}" '.format(canonicalize_identifier(c))
+
+    if ('properties', c) in md_map and md_map[('properties', c)]['sql-datatype'].startswith('timestamp'):
+        return f'CASE ' \
+               f'WHEN {column_name} < \'0001-01-01 00:00:00.000000\' ' \
+               f'OR {column_name} > \'9999-12-31 23:59:59.999999\' THEN \'9999-12-31 23:59:59.999\' ' \
```

Comment:
I think we need to use milliseconds everywhere otherwise dates like `9999-12-31 23:59:59.999123` or `0000-01-01 00:00:00.000123` will not be converted to the fallback date in milliseconds.

Reply:
I don't think I understand your comment!
Can you give an example of what you mean?

Reply:
just use milliseconds everywhere:

```
return f'CASE ' \
               f'WHEN {column_name} < \'0001-01-01 00:00:00.000\' ' \
               f'OR {column_name} > \'9999-12-31 23:59:59.999\' THEN \'9999-12-31 23:59:59.999\' ' \

```

Reply:
Ok, I'm fine with that too.

Reply:
Updated in https://github.com/transferwise/pipelinewise-tap-postgres/pull/62/commits/586aba77aaf115604a5565a81a897e7bc05f5624